lib/index.js

```diff
@@ -9,38 +9,23 @@ let
 function MqttProtocol () {
   this.config = {};
   this.protocol = 'mqtt';
-  this.isDummy = false;
   this.context = null;
-  this.connectionPool = {};
+  this.connections = new WeakMap();
+  this.connectionsById = {};
   this.server = {};
 
-  this.init = function (config, context, isDummy) {
-    if (!config) {
-      throw new Error('plugin-mqtt: A configuration parameter is required');
-    }
-
-    if (!config.port) {
-      this.isDummy = true;
-      console.error(new Error('plugin-mqtt: the \'port\' attribute, with the port to listen to, is required'));
-      return false;
-    }
-
-    if (!config.requestTopic || !config.responseTopic) {
-      this.isDummy = true;
-      console.error(new Error('plugin-mqtt: the "requestTopic" and "responseTopic" attributes are required'));
-      return false;
-    }
+  this.init = function (config, context) {
+    this.config = config || Object.assign({
```

Comment:
shouldn't it be rather `this.config = Object.assign({...}, config);` instead of `this.config = config || Object.assign({...}, config);` ?